src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -162,43 +161,26 @@ async def do_update(self, job, data):
                 if error := await self.destination_pool_error(new['pool']):
                     verrors.add('sysdataset_update.pool', error)
 
-        if new['pool'] and new['pool'] != await self.middleware.call('boot.pool_name'):
-            pool = await self.middleware.call('pool.query', [['name', '=', new['pool']]])
-            if not pool:
+        if new['pool'] == await self.middleware.call('boot.pool_name'):
```

Comment:
Do we need this conditional ? I think `pool_choices` would still work nicely if the new configured pool is `boot-pool` ?